src/figuresClasses.ts

```diff
@@ -1,19 +1,109 @@
 export interface Figure {
+  shape: string,
+  color: string,
+  getArea(): number,
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
-}
+export class Triangle implements Figure {
+  public color: Color;
+
+  public shape: Shape;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
 
-export class Circle {
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.shape = Shape.Triangle;
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
+    const sides: number[] = [a, b, c].sort(); /* Sort helps with detecting 0s
+                                      if they were not in the first position */
+
+    if ((sides[0] + sides[1] <= sides[2])
+    || (sides[2] + sides[0] <= sides[1])
+    || (sides[2] + sides[1] <= sides[0])
```

Comment:
change arrays to regular variables in all places
Even if it's easier for you to manage it's its harder to read for somebody else, you know what they contain but it might not be so obvious for someone checking it, it's better to use named variables